pkg/certificate/conditions.go

```diff
@@ -0,0 +1,111 @@
+package certificate
+
+import (
+	"time"
+
+	"github.com/openservicemesh/osm/pkg/apis/config/v1alpha2"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+const (
+	ready              v1alpha2.MeshRootCertificateConditionType = "Ready"
+	accepted           v1alpha2.MeshRootCertificateConditionType = "Accepted"
+	issuingRollout     v1alpha2.MeshRootCertificateConditionType = "IssuingRollout"
+	validatingRollout  v1alpha2.MeshRootCertificateConditionType = "ValidatingRollout"
+	issuingRollback    v1alpha2.MeshRootCertificateConditionType = "IssuingRollback"
+	validatingRollback v1alpha2.MeshRootCertificateConditionType = "ValidatingRollback"
+)
+
+const (
+	trueStatus    v1alpha2.MeshRootCertificateConditionStatus = "True"
+	falseStatus   v1alpha2.MeshRootCertificateConditionStatus = "False"
+	unknownStatus v1alpha2.MeshRootCertificateConditionStatus = "Unknown"
+)
+
+const (
+	pendingReason = "Pending"
+
+	// Accepted type
+	certificateAcceptedReason = "CertificateAccepted"
+	errorRetrievingCAReason   = "ErrorRetrievingCA"
+
+	// ValidatingRollout type
+	passiveStateValidatingReason      = "PassiveState"
+	passivelyInUseForValidatingReason = "CertificatePassivelyInUseForValidation"
+	notAcceptedValidatingReason       = "NotAccepted"
+
+	// IssuingRollout type
+	passiveStateIssuingReason      = "PassiveState"
+	passivelyInUseForIssuingReason = "CertificatePassivelyInUseForIssuing"
+	notAcceptedIssuingReason       = "NotAccepted"
+
+	// IssuingRollback type
+	isNotReadyIssuingReason      = "IsNotReady"
+	passiveStateCertIssuedReason = "PassiveStateAndCertIsBeingIssued"
+	noLongerIssuingReason        = "NoLongerIssuing"
+
+	// ValidatingRollback type
+	isNotReadyValidatingReason    = "IsNotReady"
+	passiveStateCertTrustedReason = "PassiveStateAndCertIsTrusted"
+	noLongerValidatingReason      = "NoLongerValidating"
+
+	// Ready type
+	rotationCompleteReason = "RotationComplete"
+)
+
+func setMRCCondition(mrc *v1alpha2.MeshRootCertificate, conditionType v1alpha2.MeshRootCertificateConditionType, conditionStatus v1alpha2.MeshRootCertificateConditionStatus, conditionReason string, message string) {
+	newCondition := v1alpha2.MeshRootCertificateCondition{
+		Type:    conditionType,
+		Status:  conditionStatus,
+		Reason:  string(conditionReason),
+		Message: message,
+	}
+
+	now := metav1.NewTime(time.Now())
+	newCondition.LastTransitionTime = &now
+
+	for i, condition := range mrc.Status.Conditions {
```

Comment:
@steeling @jaellio what do you think of an optimization here? Could the certificate manager maintain a map of condition names/types -> condition struct pointers for O(1) updates? The condition slice isn't huge, but multiple `setMRCCondition` calls will result in multiple iterations over the slice

Reply:
Would we recreate this mapping for each MRC event?

Reply:
No, we could store it on the certificate manager

Reply:
Yes!

Reply:
I might hold off on this optimization for a future PR.